pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -404,19 +404,39 @@ public synchronized PinotResourceManagerResponse addInstance(@Nonnull Instance i
     TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableNameWithType);
     List<String> segmentNames = new ArrayList<>();
     if (tableType == TableType.OFFLINE) {
-      for (OfflineSegmentZKMetadata segmentZKMetadata : ZKMetadataProvider.getOfflineSegmentZKMetadataListForTable(
-          _propertyStore, tableNameWithType)) {
+      for (OfflineSegmentZKMetadata segmentZKMetadata : getOfflineSegmentMetadata(tableNameWithType)) {
         segmentNames.add(segmentZKMetadata.getSegmentName());
       }
     } else {
-      for (RealtimeSegmentZKMetadata segmentZKMetadata : ZKMetadataProvider.getRealtimeSegmentZKMetadataListForTable(
-          _propertyStore, tableNameWithType)) {
+      for (RealtimeSegmentZKMetadata segmentZKMetadata : getRealtimeSegmentMetadata(tableNameWithType)) {
         segmentNames.add(segmentZKMetadata.getSegmentName());
       }
     }
     return segmentNames;
   }
 
+  @Nonnull
+  public List<OfflineSegmentZKMetadata> getOfflineSegmentMetadata(@Nonnull String tableNameWithType) {
+    Preconditions.checkArgument(TableNameBuilder.isTableResource(tableNameWithType),
```

Comment:
getTableTypeFromTableName() returns null, and that will end up throwing a null pointer exception.